### ISSUE TITLE
Use JSON or YAML for config

### ISSUE THREAD

**Issue by kojisekig** [LTR4L/ltr4l#53]
Related to #52

**Comment by kojisekig** [comment]
sample for JSON config which I'm thinking is:

```json
{
  "algorithm" : "FRankNet",
  "params" : {
    "numIterations" : 100,
    "learningRate" : 0.001,
    "optimizer" : "sgd",
    "weightInit" : "normal",
    "regularization" : {
      "regularizer" : "L2",
      "rate" : 0.01
    },
    "layers" : [
      {
        "activator" : "identity",
        "num" : 5
      },
      {
        "activator" : "sigmoid",
        "num" : 1
      }
    ],
    "model" : {
      "format" : "json",
      "file" : "franknet-model.json"
    }
  },

  "evaluation" : {
    "evaluator" : "NDCG",
    "params" : {
      "k" : 10
    }
  },

  "report" : {
    "format" : "csv",
    "file" : "franknet-report.csv"
  }
}
```

**Comment by kojisekig** [comment]
and I'd like to use Jackson to read config/model and write model because Solr uses it.

**Comment by kojisekig** [comment]
This is solved via #119
